lms/static/coffee/src/instructor_dashboard/membership.coffee

```diff
@@ -670,6 +670,564 @@ class AuthList
       success: (data) -> cb?(data)
       error: std_ajax_err => @$request_response_error.text gettext "Error changing user's permissions."
 
+class EmailSelectors
+  constructor: (@$container, @$section, params={}) ->
+    params = _.defaults params,
+      label : $container.data 'label'
+
+    template_html = $("#email-list-widget-template").html()
+    @$container.html Mustache.render template_html, params
+    @cur_column = 0
+    @$table = $("#queryTableBody")
+    @labelArray = ($container.data 'selections').split '<>'
+    @$list_selector = @$container.find 'select.single-email-selector'
+    # populate selectors
+    @$list_selector.empty()
+    @$list_endpoint = $container.data 'list-endpoint'
+    @$rolename = $container.data 'rolename'
+    @$list_selector.append $ '<option/>'
+    if this.$container.attr('data-label') == 'Select Section'
+      @load_list()
+    else if this.$container.attr('data-label') == 'Select Problem'
+      @load_list()
+    else
+      for label in @labelArray
+        @$list_selector.append $ '<option/>',
+          text: label
+
+    @$list_selector.change =>
+      $opt = @$list_selector.children('option:selected')
+      return unless $opt.length > 0
+
+      if this.$container.attr('data-label') == "Select a Type"
+        @chosen_class = $opt.text().trim()
+        if (@chosen_class == "Section")
+          @$section.find('.problem_specific').removeClass("active")
+          @$section.find('.section_specific').addClass("active")
+        else if (@chosen_class == "Problem")
+          @$section.find('.section_specific').removeClass("active")
+          @$section.find('.problem_specific').addClass("active")
+
+  get_list: (cb)->
+    $.ajax
+      dataType: 'json'
+      url: @$list_endpoint
+      data: rolename: 'instructor'
+      success: (data) -> cb? null, data['data']
+      error: std_ajax_err ->
+        cb? gettext("Error fetching problem or section data")
+
+  # load section/problem data
+  load_list: ->
+    @get_list (error, section_list) =>
+      # abort on error
+      return @show_errors error unless error is null
+      _.each section_list, (section) =>
+        @add_row(section, "section")
+        _.each section.sub, (subsection) =>
+          @add_row(subsection , "subsection")
+
+  add_row: (node, useClass) ->
+    @idArr = [node.block_type, node.block_id]
+    @idSt = @idArr.join("/")
```

Comment:
I'm not seeing `idSt` referenced outside of this method; why are we assigning it as an instance variable instead of making it local?